rstar/src/rtree.rs

```diff
@@ -294,7 +294,7 @@ where
     ///     println!("This tree contains point {:?}", point);
     /// }
     /// ```
-    pub fn iter(&self) -> RTreeIterator<T> {
+    pub fn iter(&'_ self) -> RTreeIterator<'_, T> {
```

Comment:
```suggestion
    pub fn iter(&self) -> RTreeIterator<'_, T> {
```

I think the left-hand side is generally unnecessary here to silence the lint (since `&` implies a lifetime) and I would also prefer it.

Reply:
Nice. Could you apply it uniformly to all (self) references?